lib/serial_terminal.pm

```diff
@@ -107,6 +107,9 @@ sub login {
     # after reboot and start typing the username before the console is actually
     # ready to accept it
     wait_serial(qr/login:\s*$/i, timeout => 5, quiet => 1);
+    # make sure that login prompt appears even when the console has been
+    # previously used
+    type_string(qq(\cd));
```

Comment:
I like such ideas from usage perspective. But from test perspective you could argue that if the login prompt doesn't appear, where we actually expect it, its a **really** bad bug!

So I would still be against it.

Reply:
I think it is a test case bug as it appears only in case of failure.

Reply:
so it depends on infra?

Reply:
I do not think so, the thing is that the failure happened when the test had root tty and root-serial console logged. After the failure, the test suite was rollback to the state after consoletest_setup ( so both consoles logged ). So both tty and serial consoles are still logged. For some reason, openQA expects serial console as logged out.